lib/crypto/c_src/engine.c

```diff
@@ -26,6 +26,9 @@ struct engine_ctx {
     char *id;
 };
 
+#define ERROR_Term(Env, ReasonTerm) enif_make_tuple2(Env, atom_error, (ReasonTerm))
+#define ERROR_Atom(Env, ReasonString) ERROR_Term(Env, enif_make_atom(Env,ReasonString))
```

Comment:
nit: macro parameters should have parentheses when used as arguments in the definition to avoid pitfalls.